### ISSUE TITLE
Unable to boot LEDE after installation possibly due to "U-boot environment could not be decrypted"

### ISSUE THREAD

**Issue by hhuz** [mkresin/lede#1]
I have 2 BT "Home Hub 5 - Type A" routers. I followed the v1.0 steps to switch the stock firmware to LEDE for both routers. I got `WARNING: U-boot environment could not be decrypted!` when running `prepare` for both routers. After installation, routers kept showing `CFG 06` (`VR9` prompt didn't appear). I tried both `squashfs` and `ubifs` without success. mtd0 and mtd1 for both routers are attached, mtd1 is identical for both routers.

[mtd.tar.gz](https://github.com/mkresin/BTHH5a-installimage/files/1533437/mtd.tar.gz)

Below are the installation steps:

1. `cat lede-lantiq-bthomehubv5a_ram-u-boot.asc > /dev/ttyUSB0`
2. `tftpboot lede-lantiq-xrx200-BTHOMEHUBV5A-installimage.bin; bootm`
3. Backup stock firmware (`mtd0, mtd1, mtd2, mtd3, mtd4`)
4. `prepare`
5. `sysupgrade lede-17.01.4-lantiq-xrx200-BTHOMEHUBV5A-squashfs-sysupgrade.bin` (also tried `sysupgrade lede-17.01.4-lantiq-xrx200-BTHOMEHUBV5A-ubifs-sysupgrade.bin`). This takes a short while to finish.
6. `CFG 04` appears
7. Unplug power from mains socket then plug it back
8. `CFG 06` appears, then `NAND Read OK` repeats every 30 seconds for 3 times, then it's back to `CFG 06` and repeats. Output is included below

>ROM VER: 1.1.4
CFG 06
NAND
NAND Read OK
NAND
NAND Read OK
NAND
NAND Read OK
>
>ROM VER: 1.1.4
CFG 06
NAND
NAND Read OK
NAND
NAND Read OK
NAND
NAND Read OK
...

Below is the same as above showing where 30s intervals are

>ROM VER: 1.1.4
CFG 06
NAND
NAND Read OK
--(30s)
NAND
NAND Read OK
--(30s)
NAND
NAND Read OK
--(30s)
ROM VER: 1.1.4
CFG 06
NAND
NAND Read OK
--(30s)
NAND
NAND Read OK
--(30s)
NAND
NAND Read OK
...

**Comment by mkresin** [comment]
> I have 2 BT "Home Hub 5 - Type A" routers.
> After installation, routers kept showing CFG 06 (VR9 prompt didn't appear)

I've no idea what went wrong nor why the u-boot env couldn't be decrypted for you.

I've tried both mtd0 dumps + the mtd1 dump on my HomeHub 5a and they work as expected. The env can be decrypted successfully.

All I can suggest is to restore your backup and start from scratch.

**Comment by hhuz** [comment]
I found out mtd0 of both routers were different from the backup. I restored mtd0 and mtd1 and ran `prepare` again and decryption worked so I rebooted the router (without flashing LEDE again) and it worked for both routers. I don't know why it didn't work previously, what I did differently this time is that I didn't omit `0x81000000` (which isn't always used in the OpenWrt serial installation article) when booting from tftp.
